include/binmap/graph.hpp

```diff
@@ -55,6 +55,10 @@ class Graph {
   mapping_;
   mutable std::vector<int> *distance_matrix_;
 
+  boost::unordered_map<boost::filesystem::path, int> passed_path_;
```

Comment:
From its usage, it seems to me this could be a set and not a map. Also, it's very similar to the visited variable in `scan.cpp`, may be you could call it like this too? or `visited_path`.

Reply:
You're right, that will be more coherent